### ISSUE TITLE
What would be the best repr for leafs

### ISSUE THREAD

**Issue by drvinceknight** [theref/sage-game-theory#110]
Currently:

```
sage: leaf_4 = Leaf({player_1: 3, player_2: 2})
sage: leaf_4 
(3, 2)
```

I wonder if 1 of the following would be better:

```
sage: leaf_4 
{player_1: 3, player_2: 2}
```

```
sage: leaf_4 
Extensive form game leaf with utilities given by: {player_1: 3, player_2: 2}
```

Now either of those would require us to have a different label used for the tree plot...

**Comment by hhlorrimore** [comment]
This could be done, I prefer the first option because it does mean that if we're ever calling upon a list of leaves (such as one example we have with 8 leaves), it'll have a lot less characters than the second one, but for me that's just a personal aesthetic preference. 
I'm not sure how we'll manage the labelling of the tree plot. We'd need a way that each label is read as unique, so even if for example

```
sage: leaf_1.label
(0,1)
sage: leaf_2.label
(0,1)
sage: leaf_1.label is leaf_2.label
False
```

Obviously I'm not suggesting it's done exactly like that, but that's the issue that needs to be considered for the tree to work.

**Comment by drvinceknight** [comment]
> Most classes in sage have output similar to the second one so I'll say go
> for the second representation. If we really need a short string
> representation, then that can also be implemented (but I don't see much of
> a need).

Yeah I agree that I prefer the second one, the short string representation is relevant to the graphical output. The tree is completely overloaded if we used that longer representation so we will need to manually label each leaf with a shorted string I think... (That shouldn't be a problem)

>  I'm not sure how we'll manage the labelling of the tree plot. We'd need a
> way that each label is read as unique, so even if for example
> 
>  sage: leaf_1.label
>  (0,1)
>  sage: leaf_2.label
>  (0,1)
>  sage: leaf_1.label is leaf_2.label
>  False
> 
>  Obviously I'm not suggesting it's done exactly like that, but that's the
> issue that needs to be considered for the tree to work.
> Why does it the label have to be unique? Isn't it just a string?

Yeah, the label is just a string so it's fine if two nodes/leaves have the same label. :+1:

**Comment by drvinceknight** [comment]
This has been addressed by daac8d15c501fa70ad972979046d201dfdff5347
